compiler/prelude/prelude.go

```diff
@@ -372,7 +372,7 @@ var $internalAppend = function(slice, array, offset, length) {
 
 var $equal = function(a, b, type) {
   if (type === $jsObjectPtr) {
-    return a === b;
+    return a === b || (a.hasOwnProperty("object") && a.object === undefined && b.hasOwnProperty("object") && b.object === undefined);
```

Comment:
In the `x == js.Undefined` case, `a` and `b` on this line should end up being the JS value `undefined`. But this does not happen. The issue is in `$interfaceIsEqual`, which does not correctly unwrap the JS value. It already shows with this simple test:
```go
var i interface{} = js.Undefined
println(i == js.Undefined)
```
So `$interfaceIsEqual` needs to be fixed. I think it does not even need to call `$equal` in this case, because this just boils down to `a === b` which can be inlined.